### PR TITLE
feat(app): use nanoseconds in vehicle control API

### PR DESCRIPTION
**Pull request by thonag**
## Type of this PR 

- [ ] Bug fix
- [x] Enhancement

## Description

In vehicle OS are now ns instead of ms used.

## Issue(s) related to this PR

#265
 
 
## Affected parts of the online documentation

autogenerated javadoc

## Definition of Done

- [x] You have read CONTRIBUTING.md carefully.
- [x] You have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [x] All commits are signed-off (`-s`) with your mail address of your Eclipse Account.
- [x] `origin/main` has been merged into your Fork.
- [x] New functionality is covered by unit tests or integration tests. Code coverage must not decrease.
- [x] There are no new SpotBugs warnings. 
- [x] Coding guidelines have been observed (see CONTRIBUTING.md).
- [x] All tests pass.

### CODE REVIEW

File: app/tutorials/weather-warning/src/main/java/org/eclipse/mosaic/app/tutorial/SlowDownApp.java [hunk 1]
```diff
@@ -71,7 +72,7 @@ public void onVehicleUpdated(@Nullable VehicleData previousVehicleData, @Nonnull
 
         if (strength > 0 && !hazardousArea) {
             // Reduce speed when entering potentially hazardous area
-            getOs().changeSpeedWithInterval(SPEED, 5000);
+            getOs().changeSpeedWithInterval(SPEED, (int) (5000 * TIME.MILLI_SECOND));
```
Comment (kschrab) [inline app/tutorials/weather-warning/src/main/java/org/eclipse/mosaic/app/tutorial/SlowDownApp.java:75]: remove int cast.

File: app/tutorials/example-applications/src/main/java/org/eclipse/mosaic/app/tutorial/configurableapp/VehicleConfigurationApp.java [hunk 2]
```diff
@@ -69,7 +69,7 @@ public void processEvent(Event event) throws Exception {
             int wantedSpeed = this.config.fooInteger;
             if (getOs().getVehicleData().getSpeed() > wantedSpeed) {
                 getLog().info("The current speed equals {}m/s will be slowed down to the wanted speed equals {}m/s", getOs().getVehicleData().getSpeed(), wantedSpeed);
-                getOs().slowDown(wantedSpeed, 10_000); // 10000ms -> 10s
+                getOs().slowDown(wantedSpeed, (int) (10 * TIME.SECOND));
```
Comment (kschrab) [inline app/tutorials/example-applications/src/main/java/org/eclipse/mosaic/app/tutorial/configurableapp/VehicleConfigurationApp.java:72]: remove int cast

File: fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/VehicleUnit.java [hunk 3]
```diff
@@ -169,7 +169,7 @@ public void changeLane(VehicleLaneChange.VehicleLaneChangeMode vehicleLaneChange
 
 
     @Override
-    public void slowDown(float speed, int interval) {
+    public void slowDown(float speed, long interval) {
         VehicleSlowDown vehicleSlowDown = new VehicleSlowDown(
```
Comment (kschrab) [inline fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/VehicleUnit.java:173]: Could you change the field timeInterval in VehicleSlowDown to long/nanoseconds too? For uniformity between the classes related to vehicle control.

File: fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/AbstractSumoAmbassador.java [hunk 4]
```diff
@@ -572,7 +572,7 @@ private synchronized void receiveInteraction(VehicleSlowDown vehicleSlowDown) th
             );
         }
         bridge.getVehicleControl()
-                .slowDown(vehicleSlowDown.getVehicleId(), vehicleSlowDown.getSpeed(), (int) vehicleSlowDown.getInterval());
+                .slowDown(vehicleSlowDown.getVehicleId(), vehicleSlowDown.getSpeed(), nsToMs((long) (vehicleSlowDown.getInterval() * TIME.SECOND)));
```
Comment (kschrab) [inline fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/AbstractSumoAmbassador.java:575]: change field in VehicleSlowDown to long/nanoseonds and you wont need the extra conversion here

**Review by kschrab — CHANGES_REQUESTED** [review]
Only some minor corrections requested

File: app/tutorials/weather-warning/src/main/java/org/eclipse/mosaic/app/tutorial/SlowDownApp.java (hunk 1 above)
Comment (thonag) [reply app/tutorials/weather-warning/src/main/java/org/eclipse/mosaic/app/tutorial/SlowDownApp.java]: Is implemented

File: fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/VehicleUnit.java (hunk 3 above)
Comment (thonag) [reply fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/VehicleUnit.java:173]: Is implemented

File: fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/AbstractSumoAmbassador.java (hunk 4 above)
Comment (thonag) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/AbstractSumoAmbassador.java]: Is implemented

**Comment by thonag** [comment]
Your requests have been implmented.

**Pull request merged**